DfE.FindInformationAcademiesTrusts/Services/Trust/TrustService.cs

```diff
@@ -80,4 +81,33 @@ public async Task<TrustContactsServiceModel> GetTrustContactsAsync(string uid)
         return new TrustContactsServiceModel(trustRelationshipManager, sfsoLead, accountingOfficer, chairOfTrustees,
             chiefFinancialOfficer);
     }
+    public async Task<TrustOverviewServiceModel> GetTrustOverviewAsync(string uid)
+    {
+        var academiesOverview = await academyRepository.GetAcademiesInTrustOverviewAsync(uid);
+
+        var totalAcademies = academiesOverview.Length;
+
+        var academiesByLocalAuthority = academiesOverview
+            .GroupBy(a => a.LocalAuthority)
+            .ToDictionary(g => g.Key ?? "Unknown", g => g.Count());
+
+        var totalPupilNumbers = academiesOverview.Sum(a => a.NumberOfPupils ?? 0);
+        var totalCapacity = academiesOverview.Sum(a => a.SchoolCapacity ?? 0);
```

Comment:
We could move all the logic in this method into the repository layer and have it (mostly) done by entity framework. This would also reduce the amount of data retrieved from the database to the bare essentials